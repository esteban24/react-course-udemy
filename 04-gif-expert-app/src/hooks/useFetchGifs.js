import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = ( category ) => {
    const [state, setState] = useState(({
        data: [],
        loading: true
    }));

    // In old React with classes, this is componentDidMount()
    useEffect(() => {
        getGifs( category )
            .then( images => {
                setState({
                    data: images,
                    loading: false
                });
            });
    }, [category]);

    return state;
}