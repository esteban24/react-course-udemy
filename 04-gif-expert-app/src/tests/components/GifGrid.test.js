import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('GifGrid.js test suite', () => {
    const category = "Jurassic Park";

    test('should match snapshot', () => {
        useFetchGifs.mockReturnValue({ data:[], loading: true });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should show items when images are loaded', () => {
        useFetchGifs.mockReturnValue({ data:[{ id: "1234", title: "Triceracops", url: "https://triceracops.png" }], loading: false });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should not exist <p>Loading...</p>', () => {
        useFetchGifs.mockReturnValue({ data:[{ id: "1234", title: "Triceracops", url: "https://triceracops.png" }], loading: false });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper.find('p').exists()).toBe(false);
    })

    test('should have same number of <GifGridItem /> than gifs returned', () => {
        const gifs = [{ id: "1234", title: "Triceracops", url: "https://triceracops.png" }, { id: "ABS", title: "Triceracops", url: "https://triceracops.png" }];
        useFetchGifs.mockReturnValue({ data: gifs, loading: false });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})
