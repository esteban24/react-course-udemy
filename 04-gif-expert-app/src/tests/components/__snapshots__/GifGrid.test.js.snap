// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid.js test suite should match snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Jurassic Park
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`GifGrid.js test suite should show items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Jurassic Park
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1234"
      key="1234"
      title="Triceracops"
      url="https://triceracops.png"
    />
  </div>
</Fragment>
`;
